### PR TITLE
#7516: Move to_memory_config to C++

### PR DESCRIPTION
**Pull request by eyonland**

### CODE REVIEW

**Review by ntarafdar — APPROVED** [review]
LGTM! Not that we need this but do we have any numbers for perf improvements. I think we have a few to_memory_config op tests that maybe we can loop over.

File: ttnn/cpp/pybind11/operations/core.hpp [hunk 1]
```diff
@@ -32,7 +32,9 @@ void py_module(py::module& module) {
 
     module.def(
         "reshape",
-        [](const ttnn::Tensor& tensor, const std::array<int32_t, 2>& shape) -> ttnn::Tensor { return ttnn::reshape(tensor, shape); },
+        [](const ttnn::Tensor& tensor, const std::array<int32_t, 2>& shape) -> ttnn::Tensor {
```
Comment (arakhmati) [inline ttnn/cpp/pybind11/operations/core.hpp:35]: Can you remove the lambda and use `&ttnn::reshape`?

File: ttnn/cpp/pybind11/operations/core.hpp [hunk 2]
```diff
@@ -56,6 +58,14 @@ void py_module(py::module& module) {
         "unsqueeze_to_4D",
         [](const ttnn::Tensor& tensor) -> ttnn::Tensor { return ttnn::unsqueeze_to_4D(tensor); },
         py::arg("tensor"));
+
+    module.def(
```
Comment (arakhmati) [inline ttnn/cpp/pybind11/operations/core.hpp:62]: Can you remove the lambda and use `&ttnn::operations::core::to_memory_config`?

**Review by arakhmati — APPROVED** [review]

File: ttnn/cpp/pybind11/operations/core.hpp (hunk 1 above)
Comment (eyonland) [reply ttnn/cpp/pybind11/operations/core.hpp:35]: Only by using something like this to resolve the overload.  This looks uglier to me.
static_cast<ttnn::Tensor (*)(const ttnn::Tensor&, const ttnn::Shape&)>(&ttnn::reshape),

**Pull request merged**